### PR TITLE
[JUJU-1452] Fix updating the mongo memory profile

### PR DESCRIPTION
**Pull request by wallyworld**
A patch a while back changed `mongo.EnsureServer` so that it exited early if mongo was already installed (only for snap). This was so that upgrades didn't fail. However, this breaks changing mongo config as the changes are not written out.
Here we add a new `EnsureServerStarted` method so upgrades can check if mongo is running. And the old `EnsureServer` behaviour is reinstated.

Also remove legacy code.

## QA steps

```
juju bootstrap
juju controller-config mongo-memory-profile=low
```

ssh to controller and `cat /var/snap/juju-db/common/juju-db.config`
see that `wiredTigerCacheSizeGB = 0.25`

### CODE REVIEW

File: mongo/mongo.go [hunk 1]
```diff
@@ -212,12 +212,44 @@ type EnsureServerParams struct {
...
+	}
+	if !installed {
+		return errors.NotFoundf("mongo service %q", svc.Name())
+	}
+	running, err := svc.Running()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if !running {
+		if err = svc.Start(); err != nil {
```
Comment (benhoyt) [inline mongo/mongo.go:243]: Is starting synchronous, that is, I guess we don't need to check `svc.Running()` again after starting it?

**Review by benhoyt — APPROVED** [review]
One questions, but LGTM.

File: mongo/mongo.go (hunk 1 above)
Comment (wallyworld) [reply mongo/mongo.go:243]: internally Start() calls `running, err := s.Running()`
During testing, if you change the memory value and immediately run juju status, there's a slight pause as the controller re-establishes connection and the workers restart

**Comment by manadart** [comment]
@wallyworld @benhoyt Who ran the QA steps?

**Comment by benhoyt** [comment]
@manadart I presumed Ian did. :-) As a code reviewer, I mostly don't run QA steps unless I have questions (and I assume the author did run them). I take it from your question that our team practice is for the code reviewer to always go through them?

**Pull request merged**